deployments/infrastructure-env-production.nix

```diff
@@ -37,5 +37,147 @@ with (import ./../lib.nix);
       hydra-ip = { inherit region accessKeyId; };
       cardanod-ip = { inherit region accessKeyId; };
     };
+    datadogMonitors =
+      let
+        appKey = fileContents ./../static/datadog-application.secret;
+        apiKey = fileContents ./../static/datadog-api.secret;
+        baseMonitorOptions = {
+          renotify_interval = 60;
+          include_tags = true;
+          no_data_timeframe = 10;
+          notify_audit = true;
+          notify_no_data = false;
+          require_full_window = true;
+        };
+        messages = {
+          critical = "notify @pagerduty-Datadog_Critical";
+          non_critical = "notify @pagerduty-Datadog_Non-Critical";
+        };
+      in
+      {
+        # ---------------------------------------------------------------------------
+        # cardano-node process
+        #
+
+        crit_cardano_node_process_monitor = { config, ... }:
+          {
+            inherit apiKey appKey;
+
+            name = "Critical: cardano-node process is down";
+            type = "metric alert";
+            message = messages.critical;
+            query = "\"process.up\".over(\"cardano-node\",\"process:cardano-node\").by(\"host\",\"process\").last(2).count_by_status()";
+            monitorOptions = builtins.toJSON (baseMonitorOptions // {
+              thresholds.critical = "0.2";
+            });
+          };
+
+        # ---------------------------------------------------------------------------
+        # CPU
+        #
+
+        crit_cpu_monitor = { config, ... }:
+          {
+            inherit apiKey appKey;
+
+            name = "Critical: high CPU usage";
+            type = "metric alert";
+            message = messages.critical;
+            query = "avg(last_5m):avg:system.load.norm.1{host:${config.deployment.name}.machine} by {host} > 0.9";
```

Comment:
I'm thinking we should watch for 99th percentile instead of average to get rid of most of outliers.